opacus/tests/privacy_engine_test.py

```diff
@@ -196,117 +210,89 @@ def _compare_to_vanilla(self, do_noise, do_clip, expected_match, use_closure=Fal
             if vp.grad.norm() < 1e-4:
                 # vanilla gradient is nearly zero: will match even with clipping
                 continue
 
+            atol = 1e-7 if max_steps == 1 else 1e-5
             self.assertEqual(
-                torch.allclose(vp, pp, atol=1e-8, rtol=1e-3),
+                torch.allclose(vp, pp, atol=atol, rtol=1e-3),
                 expected_match,
                 f"Unexpected private/vanilla weight match ({name})."
                 f"Should be: {expected_match}",
             )
             self.assertEqual(
-                torch.allclose(vp.grad, pp.grad, atol=1e-8, rtol=1e-3),
+                torch.allclose(vp.grad, pp.grad, atol=atol, rtol=1e-3),
                 expected_match,
                 f"Unexpected private/vanilla gradient match ({name})."
                 f"Should be: {expected_match}",
             )
 
-    def test_compare_to_vanilla(self):
-        """
-        Compare gradients and updated weights with vanilla model initialized
-        with the same seed
-        """
-        for do_noise in (True, False):
-            for do_clip in (True, False):
-                for use_closure in (True, False):
-                    with self.subTest(
-                        do_noise=do_noise, do_clip=do_clip, use_closure=use_closure
-                    ):
-                        self._compare_to_vanilla(
-                            do_noise=do_noise,
-                            do_clip=do_clip,
-                            expected_match=not (do_noise or do_clip),
-                            use_closure=use_closure,
-                        )
-
-    def test_compare_to_vanilla_accumulated(self):
+    @given(
+        # do_clip=st.booleans(),
+        # do_noise=st.booleans(),
+        # use_closure=st.booleans(),
+        # max_steps=st.sampled_from([1, 4]),
+        do_clip=st.just(False),
+        do_noise=st.just(False),
+        use_closure=st.just(False),
+        max_steps=st.sampled_from([4]),
+    )
+    @settings(deadline=None)
+    def test_compare_to_vanilla(
+        self,
+        do_clip: bool,
+        do_noise: bool,
+        use_closure: bool,
+        max_steps: int,
+    ):
         """
         Compare gradients and updated weights with vanilla model initialized
         with the same seed
         """
-        for do_noise in (False, True):
-            for do_clip in (False, True):
-                with self.subTest(do_noise=do_noise, do_clip=do_clip):
-                    self._compare_to_vanilla(
-                        do_noise=do_noise,
-                        do_clip=do_clip,
-                        expected_match=not (do_noise or do_clip),
-                    )
+        self._compare_to_vanilla(
+            do_noise=do_noise,
+            do_clip=do_clip,
+            expected_match=not (do_noise or do_clip),
+            use_closure=use_closure,
+            grad_sample_mode=self.GRAD_SAMPLE_MODE,
+            max_steps=max_steps,
+        )
```

Comment:
Fun fact: this was broken for a while too (we're supposed to call `self._compare_to_vanilla_accumulated`, which is now refactored into `self._compare_to_vanilla` with extra `max_steps` parameter)